### PR TITLE
checkbox common commponent

### PR DESCRIPTION
**Pull request by MaiUbeid**
@11janci  The previous pull request is canceled wrongly so I update your comments and make another pull request 

- renaming interest to value and all of the CSS class names to start with checkbox instead of interest
- use defaultProps for checked
#44 

code checkbox component
---------------------------------------
### create a checkbox component that contains input, label, and span to specify the user interests
-----------------------------------

# how to use :
This component will be used in the stateful component to change the state to be checked if the user check it.
1- put the this staticData
```js
const interestInfo = [
  {
    id: 1,
    interest: 'Music',
    checked: false,
    className: 'interest',
    onClick: () => console.log('change the statee!!!'),
  },
  {
    id: 2,
    interest: 'Music',
    checked: false,
    className: 'interest',
    onClick: () => console.log('change the statee!!!'),
  },
];
```
    
2- in the app.js file

```js
  import Checkbox from './components/commonComponents/Checkbox';

```
 ```js
{interestInfo.map(interest => (
          <Checkbox
            key={interest.id}
            id={interest.id}
            interest={interest.interest}
            checked={interest.checked}
            className="interest"
            onClick={interest.onClick}
          />
        ))}
```

### CODE REVIEW

File: client/src/components/common/Checkbox/index.css [hunk 1]
```diff
@@ -0,0 +1,36 @@
+.checkbox__home{
+  display: inline-flex;
+  margin: 10px;
+}
+
+.checkbox__input[type="checkbox"] {
```
Comment (11janci) [inline client/src/components/common/Checkbox/index.css]: you shouldn't need the attribute selector, the class uniquely identifies the element and the type doesn't change

File: client/src/components/common/Checkbox/index.css [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+.checkbox_label{
+  color: #ffffff;
+  padding: 10px 30px;
+  background: #3873B3;
+  border-radius: 16px;
+  cursor: pointer;
+  box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.25);
+}
+
+.checkbox_label .checkbox__input[type="checkbox"]:checked {
```
Comment (11janci) [inline client/src/components/common/Checkbox/index.css]: attribute selector can be removed

File: client/src/components/common/Checkbox/index.css [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  background: #52DE97;
+}
+
+.checkbox_label .checkbox_span .checkbox__input[type="checkbox"]:checked {
```
Comment (11janci) [inline client/src/components/common/Checkbox/index.css]: attribute selector

File: client/src/components/common/Checkbox/index.js [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    <label htmlFor={id} className="checkbox_label">
+      <input
+        id={id}
+        type="checkbox"
+        value={value}
+        className="checkbox__input"
+        checked={checked}
+        onClick={onClick}
+      />
+      <span className="checkbox_span"></span>
```
Comment (11janci) [inline client/src/components/common/Checkbox/index.js]: is this span needed?

Comment (MaiUbeid) [reply client/src/components/common/Checkbox/index.js]: yes it's needed for the component looks

**Comment by MaiUbeid** [comment]
@11janci  I removed attribute selector

File: client/src/components/common/Checkbox/index.css (hunk 3 above)
Comment (11janci) [reply client/src/components/common/Checkbox/index.css]: another thing, I don't think these selectors will match anything - the `checkbox__input` element is not within the `checkbox_span` but next to it, thus these styles will never be applied.. pls verify this and either remove or fix it

**Comment by Amoodaa** [comment]
@Fatmasiam review here?

**Pull request merged**